src/site/content/en/blog/window-controls-overlay/index.md

```diff
@@ -0,0 +1,432 @@
+
+### Being notified of geometry changes
+
+Querying the window controls overlay area with `getBoundingClientRect()` can suffice for one-off
+things like setting the correct background image based on where the window controls are, but in
+other cases, more fine-grained control is necessary. For example, a possible use case could be to
+adapt the window controls overlay based on the available space and to add a joke right in the window
+control overlay when there is enough space.
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/Unm5S2dB3szfFF2YCcFO.png", alt="Window controls overlay area on wide window displaying an additional joke (which reads Chuck Norris can find a word that rhymes with orange.)", width="800", height="73" %}
+
+You can be notified of geometry changes by subscribing to
+`navigator.windowControlsOverlay.ongeometrychange` or by setting up an event listener for the
+`geometrychange` event. This event will only fire when the window controls overlay is visible, that
+is, when `navigator.windowControlsOverlay.visible` is `true`.
+
+{% Aside %} Since this event fires frequently (comparable to how a scroll listener fires), I always
+recommend you use a
+[debounce function](https://css-tricks.com/the-difference-between-throttling-and-debouncing/#debouncing-enforces-that-a-function-not-be-called-again-until-a-certain-amount-of-time-has-passed-without-it-being-called-as-in-execute-this-function-only-if-100-milliseconds-have-passed-witho)
+so the event does not fire too often. {% endAside %}
+
+```js
+const debounce = (func, wait) => {
+  let timeout;
+  return function executedFunction(...args) {
+    const later = () => {
+      clearTimeout(timeout);
+      func(...args);
+    };
+    clearTimeout(timeout);
+    timeout = setTimeout(later, wait);
+  };
+};
+
+if ('windowControlsOverlay' in navigator) {
+  navigator.windowControlsOverlay.ongeometrychange = debounce((e) => {
+    span.hidden = e.boundingRect.width < 800;
+  }, 250);
+}
+```
+
+Rather than assigning a function to `ongeometrychange`, you can also add an event listener to
+`windowControlsOverlay` like below. You can read up on the difference between the two on
+[MDN](https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Event_handlers).
+
+```js
+navigator.windowControlsOverlay.addEventListener(
+  'geometrychange',
+  debounce((e) => {
+    span.hidden = e.boundingRect.width < 800;
+  }, 250),
+);
+```
+
+### Compatibility when running in a tab and on non-supporting browsers
+
+There are two possible cases to consider:
+
+- The case where an app is running in a browser that _does_ support Window Controls Overlay, but
+  where the app is used in a browser tab.
+- The case where an app is running in a browser that _does not_ support Window Controls Overlay.
+
+In both cases, by default the window controls overlay will display inline like regular HTML content
+and the `env()` variables' fallback values will kick in. On supporting browsers, you can also decide
```

Comment:
> On supporting browsers, you can also decide to not display the window controls overlay by checking the overlay's `visible` property and then hiding it.

This isn't something the developer would control using the readonly `visible` property. Instead, the toggle that is mentioned in the security section is intended for the user. It allows them to start off in a safe state with a `standalone` title bar, then they can toggle the WCO on using a button in the title bar and toggle it back off if they want to return back to a `standalone` window.

Reply:
I have made this clearer by separating more clearly between the window controls overlay and the HTML designated to be placed inside of the window controls overlay.